java/datastructures/lib/src/main/java/codechallenges/linkedlist/BracketLinter.java

```diff
@@ -0,0 +1,42 @@
+package codechallenges.linkedlist;
+
+import datastructures.linkedlist.Stack;
+
+public class BracketLinter {
+    public static boolean bracketLinter(String s) {
+      Stack<String> stack = new Stack<>();
+
+      for (char c : s.toCharArray()) {
+        if (c == '{' || c == '[' || c == '(') {
+          stack.push(String.valueOf(c));
+        }
+
+        if (c == '}' && !stack.isEmpty() && stack.peek().equals("{")) {
+          stack.pop();
+          if (!stack.isEmpty() && !stack.peek().equals("{")) {
```

Comment:
I'm not sure what this if statement is doing - it looks like you've already removed the correct opening bracket, so if you check for it again, you're almost always going to get an error.